src/Client.php

```diff
@@ -301,6 +323,28 @@ protected function getContext(ConnectionInterface $connection)
     }
 
     /**
+     * Derives a set of socket context options for a given secure connection.
+     *
+     * @param \Phergie\Irc\ConnectionInterface $connection
+     * @return array Associative array of context option key-value pairs
+     */
+    protected function getSecureContext(ConnectionInterface $connection)
+    {
+        $context = array();
+        if ($this->getForceIpv4Flag($connection)) {
+            $context['bindto'] = '0.0.0.0:0';
+        }
+        if ($this->getVerifyPeerFlag($connection)) {
+            $context['verify_peer'] = false;
+        }
+        if ($this->getAllowSelfSignedFlag($connection)) {
+            $context['allow_self_signed'] = true;
+        }
+        //$context = array('socket' => $context);
```

Comment:
The context we send to react socket-client for a secure connection is different.
As far as I can see passing the socket is not valid, I should have deleted that line and did that now.

Maybe I am wrong, but if I look at SecureConnector.php at react/socket-client it is doing a completely different thing than for a regular connection.